### PR TITLE
MK-983 MK-1025 - added validations and fix UI and server-side bugs.

### PR DESCRIPTION
**Pull request by kazoompa**

### CODE REVIEW

File: mica-webapp/src/main/webapp/app/entity-taxonomy-config/entity-taxonomy-config-controller.js [hunk 1]
```diff
@@ -315,15 +324,24 @@ mica.entitySfConfig
 
       MicaConfigResource.get(function (micaConfig) {
         $scope.sfOptions = {
+          validationMessage: {
+            '302': $filter('translate')('required')
+          },
           formDefaults: {
-            languages: {}
+            languages: {},
```
Comment (ymarcon) [inline mica-webapp/src/main/webapp/app/entity-taxonomy-config/entity-taxonomy-config-controller.js]: syntax error ?

File: mica-webapp/src/main/webapp/index.html [hunk 2]
```diff
@@ -212,8 +212,9 @@
 <script src="bower_components/angular-xeditable/dist/js/xeditable.js"></script>
 <script src="bower_components/angular-media-queries/match-media.js"></script>
 <script src="bower_components/sf-localized-string/dist/sf-localized-string.min.js"></script>
-<script src="bower_components/sf-obiba-file-upload/dist/sf-obiba-file-upload.min.js"></script>
+<script src="bower_components/sf-obiba-file-upload/dist/sf-obiba-file-upload.js"></script>
```
Comment (ymarcon) [inline mica-webapp/src/main/webapp/index.html]: why not "min"?

Comment (kazoompa) [inline mica-webapp/src/main/webapp/index.html]: This is to be fixed in the add-on project, I guess it was changed for debugging purposes and left in the bower/package JSON. In reality all these scripts will be minimised in Mica2 project but .min will be best for other projects.

File: mica-webapp/src/main/webapp/app/entity-taxonomy-config/entity-taxonomy-config-controller.js (hunk 1 above)
Comment (kazoompa) [inline mica-webapp/src/main/webapp/app/entity-taxonomy-config/entity-taxonomy-config-controller.js]: Corrected

File: mica-webapp/src/main/webapp/index.html (hunk 2 above)
Comment (kazoompa) [inline mica-webapp/src/main/webapp/index.html]: Corrected

**Pull request merged**